### PR TITLE
fasm: add RAMB18 features generation

### PR DESCRIPTION
**Pull request by acomodi**
This PR adds support for writing RAMB18 features from a physical and logical netlists.

It has been tested with the ram design.

I believe there are enough handling cases at the moment to start the work towards a better generalization of the FASM feature handling, by expanding the device resources database in order to hold information on how to write FASM features for the various parameters, routing bels, pips, etc.

### CODE REVIEW

File: fpga_interchange/fasm_generators/xc7.py [hunk 1]
```diff
@@ -85,6 +85,101 @@ def parse_lut_bel(lut_bel):
...
+                    fasm_features.append(fasm_feature)
+
+                elif attr in z_features:
+                    init_value = init_param.decode_integer(value)
+                    width = init_param.width
+
+                    init_str_value = "{value:0{width}b}".format(
+                        value=init_value, width=width)
+
+                    feature_value = init_str_value.replace("1", "2").replace(
```
Comment (gatecat) [inline fpga_interchange/fasm_generators/xc7.py:153]: I'm curious where the 2s are coming from? this seems to be a binary string

Comment (acomodi) [reply fpga_interchange/fasm_generators/xc7.py:153]: Hmm.. I need to double check if replacing 1s with 2s is actually permitted in this case, but I think that there actually are neater ways to implement a bit string inversion.

Comment (gatecat) [reply fpga_interchange/fasm_generators/xc7.py:153]: Oh, I missed that this was a trick for inversion.

If you want to keep using this trick, putting it in a function called something like `invert_bitstring` would make it a lot clearer.

Comment (acomodi) [reply fpga_interchange/fasm_generators/xc7.py:153]: Yeah, I guess wrapping this in a function makes sense, as bitstring inversion might be something generally useful

Comment (acomodi) [reply fpga_interchange/fasm_generators/xc7.py]: Done

**Review by gatecat — APPROVED** [review]

**Pull request merged**